go/oasis-node/cmd/consensus/consensus.go

```diff
@@ -78,24 +81,98 @@ func doSubmitTx(cmd *cobra.Command, args []string) {
 		os.Exit(1)
 	}
 
-	if err = client.SubmitTx(context.Background(), &tx); err != nil {
+	return &tx
+}
+
+func doSubmitTx(cmd *cobra.Command, args []string) {
+	if err := cmdCommon.Init(); err != nil {
+		cmdCommon.EarlyLogAndExit(err)
+	}
+
+	conn, client := doConnect(cmd)
+	defer conn.Close()
+
+	tx := loadTx()
+
+	if err := client.SubmitTx(context.Background(), tx); err != nil {
 		logger.Error("failed to submit transaction",
 			"err", err,
 		)
 		os.Exit(1)
 	}
 }
 
+func doShowTx(cmd *cobra.Command, args []string) {
+	if err := cmdCommon.Init(); err != nil {
+		cmdCommon.EarlyLogAndExit(err)
+	}
+
+	sigTx := loadTx()
+
+	var tx transaction.Transaction
+	if err := cbor.Unmarshal(sigTx.Blob, &tx); err != nil {
```

Comment:
:information_source: show_tx doesn't verify any signatures

Reply:
Correct, because otherwise you would need to pass a genesis file as well so that signatures could be verified. I'll change it so that it also displays whether the signature is valid or not if you pass the genesis file.